entity-framework/core/managing-schemas/ensure-created.md

```diff
@@ -8,18 +8,18 @@ uid: core/managing-schemas/ensure-created
 ---
 # Create and Drop APIs
 
-The EnsureCreated and EnsureDeleted methods provide a lightweight alternative to [Migrations](xref:core/managing-schemas/migrations/index) for managing the database schema. These methods are useful in scenarios when the data is transient and can be dropped when the schema changes. For example during prototyping, in tests, or for local caches.
+The <xref:Microsoft.EntityFrameworkCore.Storage.IDatabaseCreator.EnsureCreated> and <xref:Microsoft.EntityFrameworkCore.Storage.IDatabaseCreator.EnsureDeleted> methods provide a lightweight alternative to [Migrations](xref:core/managing-schemas/migrations/index) for managing the database schema. These methods are useful in scenarios when the data is transient and can be dropped when the schema changes. For example during prototyping, in tests, or for local caches.
 
-Some providers (especially non-relational ones) don't support Migrations. For these providers, EnsureCreated is often the easiest way to initialize the database schema.
+Some providers (especially non-relational ones) don't support Migrations. For these providers, `EnsureCreated` is often the easiest way to initialize the database schema.
 
 > [!WARNING]
-> EnsureCreated and Migrations don't work well together. If you're using Migrations, don't use EnsureCreated to initialize the schema.
+> `EnsureCreated` and Migrations don't work well together. If you're using Migrations, don't use `EnsureCreated` to initialize the schema.
```

Comment:
This now sounds a bit harsh. See https://github.com/dotnet/efcore/issues/26870#issuecomment-985364245. Maybe it would be good to just document how to do this as part of these changes?

Reply:
Do you mean specifically wrt testing, e.g. use EnsureCreated when testing but Migrations in production? We could amend the warning for that, though I still think it's generally valid, at least until we do #2167, no?

Reply:
If you only ever use the migrations as scaffolding (which seems common), then EnsureCreated should be fine. Its only if you edit the migrations that EnsureCreated is no longer a good choice. Contrast this to the case for EF6, where using the equivalent of EnsureCreated would (attempt to) discover your migrations and execute those, so the database created would be the same either way.